test/translator/verify_test.cc

```diff
@@ -0,0 +1,346 @@
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b010,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], 74565);
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
+}
+
+TEST(Verify, Protect011Bytchk1) {
+  nvme::GenericQueueEntryCmd nvme_cmd;
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b011,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], 74565);
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
+}
+
+TEST(Verify, Protect100Bytchk1) {
+  nvme::GenericQueueEntryCmd nvme_cmd;
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b100,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], 74565);
+  EXPECT_EQ(nvme_cmd.cdw[1], 0);
+  EXPECT_EQ(nvme_cmd.cdw[2], 536870912);
+}
+
+TEST(Verify, Protect101Bytchk1) {
+  nvme::GenericQueueEntryCmd nvme_cmd;
+  uint32_t lba = 0x12345;
+  const scsi::Verify10Command verify_cmd = {
+      .bytchk = 1,
+      .vr_protect = 0b101,
+      .logical_block_address = htonl(lba),
+      .verification_length = htons(1),
+      .control_byte = {.naca = 0},
+  };
+  const uint8_t* ptr = reinterpret_cast<const uint8_t*>(&verify_cmd);
+  translator::Span<const uint8_t> scsi_cmd =
+      translator::Span(ptr, sizeof(scsi::Verify10Command));
+  EXPECT_EQ(translator::VerifyToNvme(scsi_cmd, nvme_cmd),
+            translator::StatusCode::kSuccess);
+
+  uint8_t prchk = 0;
+  uint8_t pr_info = 0b1000 | prchk;
+  EXPECT_EQ(nvme_cmd.cdw[0], 74565);
```

Comment:
nit: this number seems kind of random, but it is expected to be `htoll(lba)` right? Since our system is big endian, technically it's the same value, but i think for clarify it would be better to have `htoll(lba)`

Reply:
this should be __byteswap_32(lba) I believe. I have added that to the test--would byteswap still work on big endian machines?

Reply:
byteswap is not necessarily what we want. So as an example, say we define `lba = 0x1a2b3c`. We want scsi to represent that value in big endian, so we set the scsi field to be `htonl(lba)` and we expect nvme to represent that value in little endian which is `htoll(lba)`. We only use byteswap when we know that we're are converting something from big endian to little endian. Technically, we can expect the nvme lba to be `__bswap_32(htonl(lba))`, but that's a little messy and that is what `htoll` is supposed to wrap

As another example, you can have a field like `network_lba = 0x1a2b3c` and set the scsi lba field to be `network_lba` and then you would expect the nvme lba to be `__bswap_32(network_lba)`